app/models/upload_cleanup_job.rb

```diff
@@ -42,26 +43,33 @@ def perform
             study_file.remove_local_copy
             Rails.logger.info "#{Time.zone.now}: cleanup for #{study_file.bucket_location}:#{study_file.id} complete"
           else
-            # remote file was not found, so attempt upload again and reschedule cleanup
-            Rails.logger.info "#{Time.zone.now}: remote file MISSING for #{study_file.bucket_location}:#{study_file.id}, attempting upload"
-            study.send_to_firecloud(study_file)
-            # schedule a new cleanup job
-            interval = retries * 2
-            run_at = interval.minutes.from_now
-            Rails.logger.info "#{Time.zone.now}: scheduling new UploadCleanupJob for #{study_file.bucket_location}:#{study_file.id}, will run at #{run_at}"
-            Delayed::Job.enqueue(UploadCleanupJob.new(study, study_file, retries), run_at: run_at)
+            # remote file was not found, so schedule a new cleanup job to check again
+            # file may be pushing in another thread, and attempting to push here creates infinite recursion if errors are encountered
+            if self.retry_count <= MAX_RETRIES
+              interval = retry_count * 2
+              run_at = interval.minutes.from_now
+              Rails.logger.info "#{Time.zone.now}: remote file MISSING for #{study_file.bucket_location}:#{study_file.id}, scheduling new UploadCleanupJob for #{run_at}"
+              Delayed::Job.enqueue(UploadCleanupJob.new(study, study_file, self.retry_count), run_at: run_at)
+            else
+              Rails.logger.error "#{Time.zone.now}: file #{study.accession}:#{study_file.bucket_location}:#{study_file.id} has failed to push to #{study.bucket_id}"
```

Comment:
Prepending all logs with `#{Time.zone.now}: ` is tedious to write, and slightly distracting to read.  Given all logs should have a timestamp, can this be abstracted to a common function?

E.g. in some other .rb file:

```
def log(message, type="info")
  if type == 'info'
    Rails.logger.info "#{Time.zone.now}: #{message}"
  elif type == 'error'
    Rails.logger.error "#{Time.zone.now}: #{message}"
    # Maybe also log to Sentry
  elif ... other log levels
end
```

Then:
```suggestion
              log("remote file MISSING for #{study_file.bucket_location}:#{study_file.id}, scheduling new UploadCleanupJob for #{run_at}")
              Delayed::Job.enqueue(UploadCleanupJob.new(study, study_file, self.retry_count), run_at: run_at)
            else
              log("file #{study.accession}:#{study_file.bucket_location}:#{study_file.id} has failed to push to #{study.bucket_id}", 'error')
```

Reply:
Good catch!  I've been slowly removing all those duplicate timestamps when I find them because they're automatically prepended in deployed environments like `staging` and `production`.  I'll remove these as well.